src/engine/directions.ts

```diff
@@ -0,0 +1,10 @@
+export default class Direction {
```

Comment:
I am not sure you need Direction to be a class here, as you are never instantiating it I think you might be able to use simply an interface e.g.
```
interface Direction {
    rowChange : number;
    colChange : number;
    }
```